### ISSUE TITLE
More cross-platform friendly

### ISSUE THREAD

**Issue by ErikReider** [rfuzzo/CP77Tools#6]
Is there any way of making this more cross-platform? I'm trying to build this in Linux. Could always use Wine but may be limiting

**Comment by rfuzzo** [comment]
Great idea! Yes that should be possible.

**Comment by sodanakin** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Github](https://trello.com/c/64mYhNeg/14-github)

**Comment by Yucked** [comment]
Since it has been moved to .NET 5, you should be able to compile it on Linux now.

**Comment by aspenluxxxy** [comment]
If you want to build this as a single binary (all you need is CP2077tools.exe), you can do this:
```
dotnet publish -r win-x64 -c Release /p:PublishSingleFile=true /p:PublishTrimmed=true /p:PublishReadyToRun=true /p:PublishReadyToRunShowWarnings=true /p:IncludeNativeLibrariesInSingleFile=true /p:IncludeNativeLibrariesForSelfExtract=true
```

Change `win-x64` to runtime id of choice.

**Comment by waitxd** [comment]
Just to mention a few caveats:
- [texconv.exe](https://github.com/microsoft/DirectXTex/releases) is used for texture conversion and there is no Linux version
- [oo2ext_7_win64.dll](http://www.radgametools.com/oodlecompressors.htm) is used for decompressing. This library is proprietary and taken from Windows release of the game. Have to find a legal way to obtain a `.so` version because there is no a Linux version of the game yet.

**Comment by ErikReider** [comment]
@waitxd we have wine. I'm able to run the game on Linux through it

**Comment by Maxzor** [comment]
> Just to mention a few caveats:
> 
>     * [texconv.exe](https://github.com/microsoft/DirectXTex/releases) is used for texture conversion and there is no Linux version
Imagemagick should be somewhat equivalent and is free software, runs on linux.
>     * [oo2ext_7_win64.dll](http://www.radgametools.com/oodlecompressors.htm) is used for decompressing. This library is proprietary and taken from Windows release of the game. Have to find a legal way to obtain a `.so` version because there is no a Linux version of the game yet.
There are reverse engineered projects made open source : https://github.com/rarten/ooz

**Comment by waitxd** [comment]
> I removed memory-mapped files, so that's one step done.

Just curious what's wrong with memory mapped files?
They are work [just fine](https://github.com/dotnet/runtime/blob/master/src/libraries/System.IO.MemoryMappedFiles/src/System/IO/MemoryMappedFiles/MemoryMappedView.Unix.cs) on Unix and improve performance on big files (less memory copying).
IIRC only named mmf is Windows-specific feature.

**Comment by rfuzzo** [comment]
yes named MMFs is the problem. RIght, I suppose we could just used normal MMFS.

**Comment by rfuzzo** [comment]
TODOS:

General:
- [ ] bring back (unnamed) Memorymapped files for reading

Remove external dependencies
- [ ] replace the cp77 oodle.dll with ooz
- [ ] replace texconv with something else for non-windows platforms
